src/main/java/org/opensearch/security/auth/UserInjector.java

```diff
@@ -66,54 +62,7 @@ public UserInjector(Settings settings, ThreadPool threadPool, AuditLog auditLog,
 
     }
 
-    public static class InjectedUser extends User {
```

Comment:
FYI: The class `InjectedUser` is no longer necessary, as it was only used to transport the value  `transportAddress` back to its direct caller, which would set it into the thread context. Afterwares, the additional fields in `InjectedUser` would not be used any more. Additionally, the `InjectdUser` was serialized to a normal `User` object.

Thus, we have replaced `Injected` user by the `Result` class below which is evaluated by the direct caller and then discarded.